tests/integrations/langchain/helpers_test.py

```diff
@@ -0,0 +1,597 @@
+
+        output = {
+            "outputs": [
+                {
+                    "generations": [
+                        [
+                            {
+                                "message": {
+                                    "kwargs": {
+                                        "usage_metadata": {
+                                            "prompt_token_count": 15,
+                                            "candidates_token_count": 8,
+                                            "total_token_count": 23,
+                                        }
+                                    }
+                                },
+                                "generation_info": {
+                                    "model_name": "gemini-2.5-pro-preview"
+                                },
+                            }
+                        ]
+                    ]
+                }
+            ]
+        }
+
+        _extract_usage_data(call, output)
+
+        assert call.summary is not None
+        assert "usage" in call.summary
+        usage = call.summary["usage"]
+        assert "gemini-2.5-pro-preview" in usage
+        assert usage["gemini-2.5-pro-preview"]["prompt_tokens"] == 15
+        assert usage["gemini-2.5-pro-preview"]["completion_tokens"] == 8
+        assert usage["gemini-2.5-pro-preview"]["total_tokens"] == 23
+
+    def test_anthropic_format(self):
+        """Test Anthropic usage format (uses input_tokens/output_tokens pattern)."""
+        call = Mock(spec=Call)
+        call.summary = None
+
+        output = {
+            "outputs": [
+                {
+                    "generations": [
+                        [
+                            {
+                                "message": {
+                                    "kwargs": {
+                                        "usage_metadata": {
+                                            "input_tokens": 25,
+                                            "output_tokens": 11,
+                                            "total_tokens": 36,
+                                        }
+                                    }
+                                },
+                                "generation_info": {
+                                    "model_name": "claude-opus-4-20250514"
+                                },
+                            }
+                        ]
+                    ]
+                }
+            ]
+        }
+
+        _extract_usage_data(call, output)
+
+        assert call.summary is not None
+        assert "usage" in call.summary
+        usage = call.summary["usage"]
+        assert "claude-opus-4-20250514" in usage
+        assert usage["claude-opus-4-20250514"]["prompt_tokens"] == 25
+        assert usage["claude-opus-4-20250514"]["completion_tokens"] == 11
+        assert usage["claude-opus-4-20250514"]["total_tokens"] == 36
+
+    def test_batch_operations_aggregation(self):
```

Comment:
Cool you have this -- do we support the batching endpoints for our other first party integrations?  If not, we should